### PR TITLE
Use Mongodb for caching

### PR DESCRIPTION
**Pull request by hieu-nguyen-tri**

### CODE REVIEW

**Comment by hieu-nguyen-tri** [comment]
Hi @JosephMontoya-TRI , this is not ready to merge yet. Could you send me the database info so I can test it out? Thanks

File: piro/mongodb.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        The epitaxies dictionary with the key is material id and the value is min epitaxy.
+    """
+    _init_mongo_client()
+    epitaxies = dict()
+    for e in _client.piro.epitaxies.find({"material_ids": target_id}):
+        for material_id in e["material_ids"]:
+            if material_id in precursor_set:
+                # If "min_epitaxy" is not in the cached entry, most likely the epitaxy calculation failed.
+                # Assign the high value so they will get discounted.
+                epitaxies[material_id] = float(e["min_epitaxy"]) if "min_epitaxy" in e.keys() else 1000000.0
```
Comment (JosephMontoya-TRI) [inline piro/mongodb.py:33]: `float(e.get("min_epitaxy", 1000000.))` is a little more idiomatic here IMO.

**Review by JosephMontoya-TRI — APPROVED** [review]
This looks good. I'd like to write a test using localhost and a test database, let's take care of that later.

File: piro/mongodb.py (hunk 1 above)
Comment (hieu-nguyen-tri) [reply piro/mongodb.py]: Ah I see, I changed it to that.

**Pull request merged**